nav2_costmap_2d/plugins/nav2_3d_static_layer.cpp

```diff
@@ -0,0 +1,226 @@
+//
+// Created by sun on 2020/10/10.
+//
+
+#include "nav2_costmap_2d/nav2_3d_static_layer.hpp"
+
+#include "octomap/octomap.h"
+#include "pcl/compression/octree_pointcloud_compression.h"
+#include "pcl/filters/voxel_grid.h"
+#include "pcl/io/pcd_io.h"
+#include "pcl/PCLPointCloud2.h"
+#include "pcl/point_cloud.h"
+#include "pcl_conversions/pcl_conversions.h"
+#include "sensor_msgs/msg/point_cloud2.hpp"
+
+using nav2_costmap_2d::NO_INFORMATION;
+using nav2_costmap_2d::LETHAL_OBSTACLE;
+using nav2_costmap_2d::FREE_SPACE;
+
+namespace nav2_costmap_2d
+{
+Nav23dStaticLayer::Nav23dStaticLayer() {
+    _map_2d = nav2_costmap_2d::Costmap2D();
+}
+Nav23dStaticLayer::~Nav23dStaticLayer(){}
+
+void
+Nav23dStaticLayer::onInitialize()
+{
+    nav2_costmap_2d::ObstacleLayer::onInitialize();
+    RCLCPP_INFO(
+            logger_,
+            "humuhumunukunukuapuaa is loading 3d static map"
+    );
+    declareParameter("enabled", rclcpp::ParameterValue(true));
+    declareParameter("topic_name", rclcpp::ParameterValue("pc2_map"));
+    declareParameter("lethal_threshold", rclcpp::ParameterValue(0.5));
+    declareParameter("map_resolution", rclcpp::ParameterValue(0.05));
+    declareParameter("voxel_leafsize", rclcpp::ParameterValue(0.2));
```

Comment:
Wouldn't it be most reasonable for the default voxel downsampling to match the costmap resolution? 

Also, the costmap resolution is accessable to you, you do not require a `map_resolution` parameter, you know what the costmap resolution is

Reply:
yeah, you're right. This parameter declaration is not necessary here.